### DISCUSSION TITLE
Fontconfig errors

### DISCUSSION THREAD

**Discussion by daladim** [hecrj/iced#861] [Q&A] [answered]
Hi,

After running the showcase on Windows without trouble (and considering using `iced` in my project because it really looked cool and simple), I have encountered the same issue as some other people related to fontconfig errors at startup on Ubuntu 20.04. \
Could you write an actual list of requirements and required dependencies in the readme?

After reading #199, #829 and #386, I have checked my installed deps:
```
$ apt list --installed | grep -E 'fontconf|pkg-conf|freetype|expat'

expat/focal,now 2.2.9-1build1 amd64 [installed]
fontconfig-config/focal,focal,now 2.13.1-2ubuntu3 all [installed,automatic]
fontconfig/focal,now 2.13.1-2ubuntu3 amd64 [installed]
libexpat1-dev/focal,now 2.2.9-1build1 amd64 [installed]
libexpat1/focal,now 2.2.9-1build1 amd64 [installed,automatic]
libexpat1/focal,now 2.2.9-1build1 i386 [installed,automatic]
libfontconfig1-dev/focal,now 2.13.1-2ubuntu3 amd64 [installed]
libfontconfig1/focal,now 2.13.1-2ubuntu3 amd64 [installed,automatic]
libfreetype-dev/focal-updates,focal-security,now 2.10.1-2ubuntu0.1 amd64 [installed]
libfreetype6-dev/focal-updates,focal-security,now 2.10.1-2ubuntu0.1 amd64 [installed]
libfreetype6/focal-updates,focal-security,now 2.10.1-2ubuntu0.1 amd64 [installed,automatic]
pkg-config/focal,now 0.29.1-0ubuntu4 amd64 [installed]
ruby-pkg-config/focal,focal,now 1.4.0-1 all [installed,auto-removable]
```

However, I still have quite a lot of errors when I run `cargo run --package tour`, such as 
```
...
Fontconfig error: "/etc/fonts/conf.d/90-synthetic.conf", line 5: invalid attribute 'translate'
Fontconfig error: "/etc/fonts/conf.d/90-synthetic.conf", line 5: invalid attribute 'selector'
Fontconfig error: "/etc/fonts/conf.d/90-synthetic.conf", line 6: invalid attribute 'xmlns:its'
Fontconfig error: "/etc/fonts/conf.d/90-synthetic.conf", line 6: invalid attribute 'version'
Fontconfig error: Cannot load default config file
```

What am I missing? Is there an up-to-date readme where I could check my config?

Thanks

**Comment by 13r0ck** [comment]
Not sure, but try changing the iced renderer to OpenGl that might remove the errors
```
iced = { version = "0.3", features=["glow"]}
```

**Comment by daladim** [comment, marked as answer]
Thanks. Adding `"glow"` in the required features (in `Cargo.toml`) works for some examples (such as `todos`), but not for others.

Here is the error I have when I try to build `tour` (with the `glow` feature) :
```
$ cargo run --package tour
   Compiling [...]
   Compiling tour v0.1.0 (/home/daladim/src/external/iced/examples/tour)
error[E0277]: the trait bound `iced_glow::backend::Backend: iced_graphics::backend::Image` is not satisfied
   --> examples/tour/src/main.rs:675:5
    |
675 |     Container::new(
    |     ^^^^^^^^^^^^^^ the trait `iced_graphics::backend::Image` is not implemented for `iced_glow::backend::Backend`
    |
    = note: required because of the requirements on the impl of `iced_native::widget::image::Renderer` for `iced_graphics::renderer::Renderer<iced_glow::backend::Backend>`
    = note: required because of the requirements on the impl of `From<Image>` for `iced_native::element::Element<'_, _, iced_graphics::renderer::Renderer<iced_glow::backend::Backend>>`
    = note: required because of the requirements on the impl of `Into<iced_native::element::Element<'_, _, iced_graphics::renderer::Renderer<iced_glow::backend::Backend>>>` for `Image`
    = note: required by `iced_native::widget::container::Container::<'a, Message, Renderer>::new`

error: aborting due to previous error

For more information about this error, try `rustc --explain E0277`.
error: could not compile `tour`

To learn more, run the command again with --verbose.
```

Anyway, this workaroung currently works for me (I don't need `Image`s in my app...yet), so you may close this issue, unless you'd like to dig the issue so that we could have a nice and perfect readme for Linux :-p

Thanks

**Comment by daladim** [comment]
For future reference, for those who will stumble on this issue later: I'm using iced at commit `4b8ba8309f6645cdcb5bd605f8dd88097f8ee5a7: Merge pull request #825 from tarkah/feat/window-visibility`

**Comment by 13r0ck** [comment]
iced_glow doesn't yet support images nor SVGs. 

We totally can add the deps list to the README, I just don't think anyone knows exactly what dep fixed the Fontconfig error. 
With that error it should still compile and run, it just takes a few minutes to start

**Comment by hecrj** [comment]
You can disable the `default_system_font` feature, instead of using `glow`.

In any case, these errors should only happen when the linked version of `fontconfig` does not match your system version. What does `pkg-config` say when you ask for `fontconfig`?

**Reply by qwerty2501** [reply]
Hi.
I tried running example of iced on Ubuntu 20.04, then I faced errors that is same https://github.com/iced-rs/iced/discussions/861#discussion-3349762.
Thank you for infomation of disable `default_system_font`. It works.

However, I guess the best is not workaround, it should be able to run iced without disable `default_system_font` on Ubuntu 20.04.
Here is `plg-config` result in my environment.
```bash
> pkg-config fontconfig --modversion
2.13.1
```
apt list
```bash
> apt list --installed | grep -E 'fontconf|pkg-conf|freetype|expat'

WARNING: apt does not have a stable CLI interface. Use with caution in scripts.

fontconfig-config/focal,focal,now 2.13.1-2ubuntu3 all [インストール済み、自動]
fontconfig/focal,now 2.13.1-2ubuntu3 amd64 [インストール済み]
freetype2-demos/focal-updates,focal-security,now 2.10.1-2ubuntu0.1 amd64 [インストール済み]
libexpat1-dev/focal,now 2.2.9-1build1 amd64 [インストール済み、自動]
libexpat1/focal,now 2.2.9-1build1 amd64 [インストール済み、自動]
libexpat1/focal,now 2.2.9-1build1 i386 [インストール済み、自動]
libfontconfig1-dev/focal,now 2.13.1-2ubuntu3 amd64 [インストール済み]
libfontconfig1/focal,now 2.13.1-2ubuntu3 amd64 [インストール済み、自動]
libfontconfig1/focal,now 2.13.1-2ubuntu3 i386 [インストール済み、自動]
libfreetype-dev/focal-updates,focal-security,now 2.10.1-2ubuntu0.1 amd64 [インストール済み]
libfreetype6-dev/focal-updates,focal-security,now 2.10.1-2ubuntu0.1 amd64 [インストール済み]
libfreetype6/focal-updates,focal-security,now 2.10.1-2ubuntu0.1 amd64 [インストール済み、自動]
libfreetype6/focal-updates,focal-security,now 2.10.1-2ubuntu0.1 i386 [インストール済み、自動]
pkg-config/focal,now 0.29.1-0ubuntu4 amd64 [インストール済み]
```

BTW:
I would like to cooperate.
Should I to do something?

**Reply by qwerty2501** [reply]
It may have been caused by font-kit.
It's just like https://github.com/servo/font-kit/issues/181
